### PR TITLE
Remove Python 3.6

### PR DESCRIPTION
**Pull request by Kludex**
Python 3.6 EOL is 23 Dec 2021, and this PR ~starts the process of removing it~ removes it. 

Reference: https://endoflife.date/python

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 **Documentation**: [https://www.uvicorn.org](https://www.uvicorn.org)
 
-**Requirements**: Python 3.6+ (For Python 3.5 support, install version 0.8.6.)
+**Requirements**: Python 3.7+ (For Python 3.6 support, install version 0.15.0)
```
Comment (Kludex) [inline README.md:16]: I've removed the 3.5 info. ~15 daily downloads for Python 3.5 and this information is only relevant if you want to start using uvicorn on a Python 3.5 project.

Reference: https://pypistats.org/packages/uvicorn

File: uvicorn/_handlers/http.py [hunk 2]
```diff
@@ -88,12 +88,7 @@ def retrieve_exception(task: asyncio.Task) -> None:
 
 
 def _get_current_task() -> asyncio.Task:
-    try:
-        current_task = asyncio.current_task  # type: ignore
-    except AttributeError:  # pragma: no cover
-        # Python 3.6.
-        current_task = asyncio.Task.current_task  # type: ignore
-
+    current_task = asyncio.current_task
```
Comment (graingert) [inline uvicorn/_handlers/http.py]:
```suggestion
    task = asyncio.current_task()
```

File: uvicorn/_handlers/http.py [hunk 3]
```diff
@@ -88,12 +88,7 @@ def retrieve_exception(task: asyncio.Task) -> None:
...
     task = current_task()
```
Comment (graingert) [inline uvicorn/_handlers/http.py]:
```suggestion
```

**Comment by Kludex** [comment]
Closed in favor of #1261

**Pull request closed without merging**